src/java.desktop/macosx/classes/sun/lwawt/LWWindowPeer.java

```diff
@@ -736,6 +736,9 @@ public void notifyReshape(int x, int y, int w, int h) {
             handleMove(x, y, true);
         }
         if (tResized || pResized || invalid || isNewDevice) {
+            if (CGraphicsDevice.usingMetalPipeline() && invalid) {
+                replaceSurfaceData();
+            }
```

Comment:
I think fix can be moved to the 729 line "if (pResized || isNewDevice || invalid)". Looks like it is a bug even in the OGL case, if "pResized == false" and window is not resized but in the insets were changed we should update the surface.

BTW I think the updateMinimumSize() should be called as well since the minimum/maximum size depends on the insets and NSWindow frame/contentRect.

Reply:
Yes, good idea. Corrected